### PR TITLE
Use operand.Display property when printing method group types to diagnostics

### PR DESCRIPTION
**Pull request by OmarTawfik**
Follow up on #14463
@dotnet/roslyn-compiler 
/cc @AlekseyTs

### CODE REVIEW

**Comment by AlekseyTs** [comment]
LGTM

**Comment by OmarTawfik** [comment]
@dotnet-bot test linux_debug_prtest please
@dotnet-bot test windows_eta_open_prtest please

**Comment by OmarTawfik** [comment]
@dotnet-bot test linux_debug_prtest please

File: src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs [hunk 1]
```diff
@@ -1784,9 +1784,7 @@ private BoundExpression BindCastCore(ExpressionSyntax node, BoundExpression oper
                 ImmutableArray<MethodSymbol> originalUserDefinedConversions = conversion.OriginalUserDefinedConversions;
                 if (originalUserDefinedConversions.Length > 1)
                 {
-                    // Method groups always have null types
-                    var operandType = (operand.Kind == BoundKind.MethodGroup) ? (object)MessageID.IDS_SK_METHOD.Localize() : operand.Type;
-
+                    var operandType = operand.Type ?? operand.Display;
```
Comment (gafter) [inline src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs]: `operand.Display` already returns the type when it is non-`null`. You should be able to just use `operand.Display`.

**Pull request merged**